worker/caasapplicationprovisioner/application.go

```diff
@@ -595,14 +597,23 @@ func (a *appWorker) refreshApplicationStatus(app caas.Application, appLife life.
 }
 
 func (a *appWorker) ensureScale(app caas.Application) error {
-	desiredScale, err := a.unitFacade.ApplicationScale(a.name)
-	if err != nil {
-		return errors.Annotatef(err, "fetching application %q desired scale", a.name)
+	var err error
+	var desiredScale int
+	switch a.life {
+	case life.Alive:
+		desiredScale, err = a.unitFacade.ApplicationScale(a.name)
+		if err != nil {
+			return errors.Annotatef(err, "fetching application %q desired scale", a.name)
+		}
+	case life.Dying, life.Dead:
```

Comment:
default?

Actually, do we even need this if all we do is set desiredScale to 0 which it will be by default anyway

Reply:
yeah really default should handle new enum values.